lib/Items/ApplicationDockItem.vala

```diff
@@ -414,19 +418,45 @@ namespace Plank
 				item.activate.connect (() => pin_launcher ());
 				items.add (item);
 			}
-			
+
 			var event_time = Gtk.get_current_event_time ();
 			if (is_running () && window_count > 0) {
 				var item = create_menu_item ((window_count > 1 ? _("_Close All") : _("_Close")), "window-close-symbolic;;window-close");
 				item.activate.connect (() => WindowControl.close_all (App, event_time));
 				items.add (item);
 			}
-			
+
+			if (!is_running () && switcheroo_control != null && switcheroo_control.has_dual_gpu) {
+
+				print("launcher: %s\n", Prefs.Launcher);
+
+				var app_info = new DesktopAppInfo.from_filename(Filename.from_uri (Prefs.Launcher));
+
+				bool prefers_non_default_gpu = app_info.get_boolean ("PrefersNonDefaultGPU");
+
+				string gpu_name = switcheroo_control.get_gpu_name (prefers_non_default_gpu);
+
+				string label = _("Open with %s Graphics").printf (gpu_name);
+
+				var item = new Gtk.MenuItem.with_mnemonic (label);
+				items.add(item);
+
+				item.activate.connect (() => {
+					try {
+						var context = new AppLaunchContext ();
+						switcheroo_control.apply_gpu_environment (context, prefers_non_default_gpu);
+						app_info.launch (null, context);
```

Comment:
Instead of launching the `app_info` manually here, I feel it would be better to do something like:
```
System.get_default ().launch (File.new_for_uri (Prefs.Launcher));
```

The `launch` method will probably need adapting to accept a new boolean argument that indicates that the default GPU should be swapped, which can default to false.

But this way, we can make sure that applications are launched in the same way via the `System` class.